vault/external_tests/raft/raft_binary/raft_test.go

```diff
@@ -47,3 +56,377 @@ func TestRaft_Configuration_Docker(t *testing.T) {
 	}
 	rafttest.Raft_Configuration_Test(t, cluster)
 }
+
+// removeRaftNode removes a node from the raft configuration using the leader client
+// and removes the docker node.
+func removeRaftNode(t *testing.T, node *docker.DockerClusterNode, client *api.Client, serverID string) {
+	t.Helper()
+	_, err := client.Logical().Write("sys/storage/raft/remove-peer", map[string]interface{}{
+		"server_id": serverID,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	node.Cleanup()
+	// TODO: remove from the cluster nodes as well.
+}
+
+// stabilizeAndPromote makes sure the given node ID is among the voters using
+// autoPilot state
+func stabilizeAndPromote(t *testing.T, client *api.Client, nodeID string) {
+	t.Helper()
+	deadline := time.Now().Add(2 * autopilot.DefaultReconcileInterval)
+	failed := true
+	var state *api.AutopilotState
+	var err error
+	for time.Now().Before(deadline) {
+		state, err = client.Sys().RaftAutopilotState()
+		require.NoError(t, err)
+		if state != nil && state.Servers != nil && state.Servers[nodeID].Status == "voter" {
+			failed = false
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	if failed {
+		t.Fatalf("autopilot failed to promote node: id: %#v: state:%# v\n", nodeID, state)
+	}
+}
+
+// stabilize makes sure the cluster is in a healthy state using autopilot state
+func stabilize(t *testing.T, client *api.Client) {
+	t.Helper()
+	deadline := time.Now().Add(2 * autopilot.DefaultReconcileInterval)
+	healthy := false
+	for time.Now().Before(deadline) {
+		state, err := client.Sys().RaftAutopilotState()
+		require.NoError(t, err)
+		if state.Healthy {
+			healthy = true
```

Comment:
Did you want to add a `break` here? Otherwise we'll end up waiting the full `deadline` before returning.